strawberry/flask/views.py

```diff
@@ -42,20 +42,32 @@ def dispatch_request(self):
             template = render_graphiql_page()
             return self.render_template(template=template)
 
-        if request.content_type.startswith("multipart/form-data"):
+        using_get: bool = request.method == "GET"
+        # GET requests doesn't have content types
+        # https://stackoverflow.com/a/16693884/2989034
+        if request.content_type is not None and request.content_type.startswith(
+            "multipart/form-data"
+        ):
             operations = json.loads(request.form.get("operations", "{}"))
             files_map = json.loads(request.form.get("map", "{}"))
 
             data = replace_placeholders_with_files(operations, files_map, request.files)
 
         else:
-            data = request.json  # type: ignore
+            attr = "args" if using_get else "json"
+            data = getattr(request, attr)
+
+        # Shorthand for using it
 
         try:
             request_data = parse_request_data(data)
         except MissingQueryError:
             return Response("No valid query was provided for the request", 400)
 
+        # Move to the @mutation annotation later?
+        if using_get and "mutation" in request_data.query:
+            return Response("GET requests don't support mutations", 400)
```

Comment:
not sure this it the best check for this, mutation might appear in a random field of the query :)

I think we might need to do something more clever here and change our `execute` and `execute_sync` functions on the schema to accept a list of operation type they can allow.

so then in dispatch_request we do something like this:


```
        result = self.schema.execute_sync(
            request_data.query,
            variable_values=request_data.variables,
            context_value=context,
            operation_name=request_data.operation_name,
            root_value=self.get_root_value(),
            allowed_operation_types=[QUERY] if get else [MUTATION, QUERY, SUBSCRIPTION]
        )
```

I'm open to other suggestions :) /cc @BryceBeagle for more ideas 😊

Reply:
Yeah, as `request_data.query` is a `str`, I think using `in` is probably pretty unsafe. I don't have any other better solutions though

Reply:
Currently made use of the `parse` function from `graphql-core` to retrieve the operation type. https://github.com/strawberry-graphql/strawberry/pull/1759/files#diff-88aa6fd17e4c6feac6e7152ebd3f2b8f972544c444a071b550e4d23061b97a3fR76